### PR TITLE
PD-1939 Dark Mode text input & minor Button fixes

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Adds Dark Mode support for `text-input`

Additionally adds a `propOverride` parameter to `useUpdatedBaseFontSize`

🎟 _Jira ticket:_ [PD-1939](https://jira.mongodb.org/browse/PD-1939)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: website/pages/component/code/guidelines.mdx [hunk 1]
```diff
@@ -42,8 +42,8 @@ import { mq } from 'utils/mediaQuery';
   >
     install the Figma Code Highlighter plugin
   </Link>
-  . This will emulate similar highlighting found in production. Use the
-  atomOneLight highlighter theme in light mode; use Agate for dark mode.
+  . This will emulate similar highlighting found in production. Use the atomOneLight
+  highlighter theme in light mode; use Agate for dark mode.
```
Comment (TheSonOfThomp) [inline website/pages/component/code/guidelines.mdx:46]: Just a prettier fix

File: .changeset/bright-kings-peel.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/button': major
```
Comment (shaneeza) [inline .changeset/bright-kings-peel.md:2]: 👍🏽

File: packages/text-input/src/TextInput.spec.tsx [hunk 3]
```diff
@@ -233,7 +234,10 @@ describe('packages/text-input', () => {
   });
 
   describe('when the "sizeVariant" is "large"', () => {
-    test('check if font-size is 18px', () => {
+    // TODO: This type of check should be done with a visual regression test
```
Comment (shaneeza) [inline packages/text-input/src/TextInput.spec.tsx:237]: 👍🏽

**Review by shaneeza — COMMENTED** [review]
🔥 Nice! I like that you separated styles and types. This file was pretty large. 

A few things:

- [ ] Should you still be able to focus when the input is `disabled`? Currently you can focus on it. 
- [ ] I still see the focus style when my I tab to another element.
- [ ] Visually, one thing that stood out was the spacing between the label and the input and the error message and the input but this could be discussed during design QA.

Screen Shot 2022-04-18 at 1 56 44 PM

**Review by shaneeza — APPROVED** [review]
✨ looks great!!

**Pull request merged**